backend/src/v2/driver/driver.go

```diff
@@ -179,9 +184,27 @@ func RootDAG(ctx context.Context, opts Options, mlmd *metadata.Client) (executio
 	if err != nil {
 		return nil, err
 	}
+
+	paramValues := opts.RuntimeConfig.GetParameterValues()
+
+	for paramName, paramDef := range opts.Component.GetInputDefinitions().GetParameters() {
+		if _, ok := paramValues[paramName]; !ok && paramDef.IsOptional {
+			if paramValues == nil {
+				paramValues = map[string]*structpb.Value{}
+			}
+
+			// Set the defaults for optional parameters not provided in the pipeline run
+			if paramDef.DefaultValue != nil {
+				paramValues[paramName] = paramDef.DefaultValue
```

Comment:
it seems to me like this case is handled [here](https://github.com/kubeflow/pipelines/blob/cc1c435f1e06aad3e9b83e57768512a63460b15b/backend/src/v2/compiler/argocompiler/argo.go#L75) right?  since you can only have a default value if it's optional (so the optional check is implicit?) 

I find it weird we're doing this partly in api server and partly in driver, maybe we ought to just do it all in driver? wdyt?

Reply:
Good find! Not sure why I got it in my head that this was needed.